Added QuartzPen

verbosus opened the pull request:
FontTools supports drawing on macOS via the [`CocoaPen`](https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/pens/cocoaPen.py), which transforms a path into an `NSBezierPath`. However, for lower-level drawing on macOS, animation with CoreAnimation, and other use-cases, it’s useful to have outlines represented as `CGPath` instead of `NSBezierPath`. RoboFab used to have a [`QuartzPen`](https://github.com/robotools/robofab/blob/master/Lib/robofab/pens/quartzPen.py) but it’s no longer functional under Python 3. This PR adds a `QuartzPen` back to FontTools, together with some unit tests that only run when PyObjC is installed.

anthrotype reviewed (approved):
Grazie Antonio! LGTM

verbosus commented:
Wow, Travis really does take forever. A few minor improvements:

- I added tests for `CocoaPen` too, since it didn’t have any;
- I removed an unused import in the `QuartzPen` tests;

I think it would be nice if the Travis osx build environment installed `pyobjc` by default, so the tests actually run there, but I don’t know how to do that.

anthrotype commented:
> it would be nice if the Travis osx build environment installed pyobjc by default

sure, but maybe better to do that after we hafe completed the migration to Github Actions (#2100)

anthrotype commented:
can you please try to rebase off master? I just merged GH Actions PR

verbosus commented:
Done, I’ve rebased the PR off of ae07ce6e554f5b4a2d6e7a6524f5a1307b307bcd and squashed all my commits in one.
